docs/user-guide/timeouts/timeouts-main.md

```diff
@@ -0,0 +1,331 @@
+and 3. Non-SMT Timeouts (Type 1.) should be reported to Certora. 
+
+(timeout_causes)=
+# What Causes Timeouts?
+
+As a first step towards resolving a timeout, we need to diagnose its root
+causes. In our experience so far, the following are some of the most common
+reasons for SMT timeouts:
+
+ - Non-trivial amount of nonlinear arithmetic
+ - Very high path count
+ - High storage/memory complexity
+
+This list is not exhaustive, but the majority of timeouts we have observed so
+far can be traced back to one or more of these causes. While these are not the
+only sources of complexity, they should work well as general estimates of
+complexity. For instance, linear arithmetic usually only becomes a problem when
+the input program is rather large, which is also indicated by path count in most
+practical cases.
+
+
+## Complexity Feedback from Certora Prover
+
+Certora Prover provides some help with diagnosing timeouts. We present these
+features in this section.
+
+### Difficulty Statistics
+
+Certora Prover provides statistics on the problem sizes it encounters. 
+These statistics are structured according to the timeout reasons given above.
+
+Currently, the Prover tracks the following statistics:
+ - Path count
+ - Nonlinear operations count
+ - Memory/storage complexity
+
+For a very short summary we give one summarizing number for each of the
+statistics, along with a LOW/MEDIUM/HIGH statement. This occurs as an INFO
+message in the Global Problems pane of the Prover reports.
+
+The exact classifications are made from experience with these statistics.
+ - LOW: unlikely to be a reason for a timeout
+ - MEDIUM: might be a reason for a timeout; the timeout might also be a result
+   of the combined complexity with other measures
+ - HIGH: likely to be a reason for a timeout, even if it is the only aspect of
+   the verification problem that shows high complexity
+
+These categories map to intervals as follows.
+
+|    | LOW | MEDIUM | HIGH |
+|----|-----|--------|------|
+| Path count | 0 to 2<sup>20</sup> | 20<sup>20</sup> to 2<sup>80</sup> | > 2<sup>80</sup> |
+| Nonlinear operations | 0 to 10 | 10 to 30 | > 30 |
+
+```{note}
+All heuristics mentioned here are might be improved over time, so 
+consider them subject to change.
+```
+
+(timeout_tac_reports)=
+### Timeout TAC Reports
+
+For each verification item, there is a TAC graph linked in the verification
+report. In case of a timeout this graph contains information one which parts of
+the program were part of the actual timeout, and which were already solved
+successfully. It also contains statistics on the above-described timeout causes.
+
+Find more documentation on TAC reports in general [here](tac-reports).
+
+In the timeout case, the TAC reports contain some additional information that
+should help with diagnosing the timeout.
+
+(timeout_prevention)=
+# Timeout Prevention
+
+Timeout prevention approaches fall into these categories.
```

Comment:
`[just a joke]` I was expected to see "do not run the prover". But really just a joke, because I think we have improved a lot with the performance :)

Reply:
🤪